oak/server/rust/src/asylo_alloc.rs

```diff
@@ -0,0 +1,34 @@
+use core::alloc::{GlobalAlloc, Layout};
+
+extern "C" {
+    // See https://www.sourceware.org/newlib/libc.html#malloc.
+    fn malloc(nbytes: usize) -> *mut u8;
+    fn memalign(align: usize, nbytes: usize) -> *mut u8;
+    fn free(aptr: *mut u8);
+}
+
+// Aligment is defined in the Intel SGX SDK.
+// See https://github.com/intel/linux-sgx/blob/master/sdk/tlibc/stdlib/malloc.c#L541.
+const MIN_ALIGN: usize = 8;
+
+/// A global allocator that links directly to the primitives exposed by Asylo, which in turns relies
+/// on [newlib](http://www.sourceware.org/newlib/libc.html) for standard `malloc` and `free` calls.
+pub struct System;
+
+// Inspired by https://github.com/apache/incubator-mesatee-sgx/blob/master/sgx_alloc/src/lib.rs.
+unsafe impl GlobalAlloc for System {
+    #[inline]
+    unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        // If the alignment requested by the caller is too small, fall back on regular malloc.
+        if layout.align() <= MIN_ALIGN && layout.align() <= layout.size() {
+            malloc(layout.size())
```

Comment:
Is this arm needed?  (i.e. would `memalign` end up doing the same as `malloc` anyway?)

Reply:
See updated comment (it is a fast path optimization).